ios/RNSScreenStack.mm

```diff
@@ -270,11 +270,7 @@ - (void)didMoveToWindow
 - (void)maybeAddToParentAndUpdateContainer
 {
   BOOL wasScreenMounted = _controller.parentViewController != nil;
-#ifdef RCT_NEW_ARCH_ENABLED
   BOOL isScreenReadyForShowing = self.window;
-#else
-  BOOL isScreenReadyForShowing = self.window && _hasLayout;
```

Comment:
Have you checked `git blame`, why the `&& _hasLayout` was there? Don't we break any other behaviour?

Reply:
I found this: https://github.com/software-mansion/react-native-screens/pull/600 and it's related to another problem I've been investigating lately by coincidence 😄 From what I've tested it works well after removing hasLayout (both test cases).